### PR TITLE
8259922 MethodHandles.collectArguments does not throw IAE if pos is outside the arity range

### PR DESCRIPTION
**Pull request by DasBrain**
Add explicit range check to `MethodHandles.collectArgumentsCheck`.  
Added test case that exercises all cases.

This is a behavioral change, from throwing an unspecified exception to throwing an IllegalArgumentException, as specified.  
No spec change needed, as the IllegalArgumentException is already specified to be thrown in those cases.

Feel free to suggest a better place for the tests.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8259922](https://bugs.openjdk.java.net/browse/JDK-8259922): MethodHandles.collectArguments does not throw IAE if pos is outside the arity range

### Reviewers
 * [Mandy Chung](https://openjdk.java.net/census#mchung) (@mlchung - **Reviewer**)

### Download
`$ git fetch https://git.openjdk.java.net/jdk pull/2171/head:pull/2171`
`$ git checkout pull/2171`

### CODE REVIEW

**Review by mlchung — COMMENTED** [review]
Thanks for fixing this.  I can sponsor this for you.

File: test/jdk/java/lang/invoke/8259922/TestMethodHandlesCollectArgs.java [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+
+import org.testng.annotations.Test;
+
+import java.lang.invoke.MethodHandle;
+import java.lang.invoke.MethodHandles;
+import static java.lang.invoke.MethodType.methodType;
+
+import static org.testng.Assert.*;
+
+public class TestMethodHandlesCollectArgs {
```
[the remark this answers is not in the document]
Comment (DasBrain) [reply test/jdk/java/lang/invoke/8259922/TestMethodHandlesCollectArgs.java:37]: Yeah, still learning where to put tests.

**Review by mlchung — APPROVED** [review]
Looks good.  What tests have you ran?

**Comment by DasBrain** [comment]
On the latest commit, only my own test.  
On the previous commit (4f74e2d) I did run the full tier1.

Currently a full test tier1 run on my machine and github actions is in progress.

**Comment by DasBrain** [comment]
Tests did run successfully on my machine (win 64) and github actions.
Let me know if there should be additional tests that I should run.

**Pull request closed without merging**